src/components/RequiredComponentProps.ts

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export interface RequiredComponentProps<T> {
+  className: string;
+  ref: React.LegacyRef<T>;
```

Comment:
Should we use legacyRef? I'm guessing this is for legacy API?

Reply:
Shouldn't we add a `style` attribute also? With type = `CSSProperties`

Reply:
At some point we need to include all the WAI-ARIA attributes as well 😛

Reply:
Should we make an efforto to add WAI-ARIA to all components at once?

Reply:
probably a good idea